test/extractor/MappingExtractor.test.ts

```diff
@@ -0,0 +1,123 @@
+    expect(mappings).toContainEqual(expectedMapping);
+
+    expect(element.processedPaths).toContain('mapping[4].identity');
+    expect(element.processedPaths).toContain('mapping[4].map');
+  });
+
+  it('should extract mappings with rules from multiple elements', () => {
+    const catElement = elements[1]; // Observation.category
+    const focusElement = elements[2]; // Observation.focus
+    const mappings = MappingExtractor.process(looseSD, elements, defs);
+    const expectedMapping = new ExportableMapping('FirstMapping');
+    expectedMapping.source = 'MyObservation';
+    const mappingRule = new ExportableMappingRule('focus');
+    mappingRule.map = 'Observation.otherFocus';
+    expectedMapping.rules.push(mappingRule);
+
+    expect(mappings).toHaveLength(4); // All mappings are created
+    expect(mappings).toContainEqual(expectedMapping);
+
+    expect(catElement.processedPaths).toContain('mapping[0].identity');
+    expect(catElement.processedPaths).toContain('mapping[0].map');
+    expect(catElement.processedPaths).toContain('mapping[0].comment');
+    expect(catElement.processedPaths).toContain('mapping[0].language');
+    expect(focusElement.processedPaths).toContain('mapping[4].identity');
+    expect(focusElement.processedPaths).toContain('mapping[4].map');
+    expect(focusElement.processedPaths).toContain('mapping[5].identity');
+    expect(focusElement.processedPaths).toContain('mapping[5].map');
+    expect(focusElement.processedPaths).toContain('mapping[5].comment');
+    expect(focusElement.processedPaths).toContain('mapping[5].language');
+  });
+
+  it('should extract mappings with rules from the top level element', () => {
+    const element = elements[0]; // Observation
+    const mappings = MappingExtractor.process(looseSD, elements, defs);
+    const expectedMapping = new ExportableMapping('TopLevelMapping');
+    expectedMapping.source = 'MyObservation';
+    const mappingRule = new ExportableMappingRule('');
+    mappingRule.map = 'Observation';
+    expectedMapping.rules.push(mappingRule);
+
+    expect(mappings).toHaveLength(4); // All mappings are created
+    expect(mappings).toContainEqual(expectedMapping);
+
+    expect(element.processedPaths).toContain('mapping[4].identity');
+    expect(element.processedPaths).toContain('mapping[4].map');
+  });
+
+  it('should not extract mappings that do not exist on the top level structure definition', () => {
+    const element = elements[3]; // Observation.status
+    const mappings = MappingExtractor.process(looseSD, elements, defs);
+    const unexpectedMapping = new ExportableMapping('RogueMapping');
+    unexpectedMapping.source = 'MyObservation';
+    const mappingRule = new ExportableMappingRule('status');
+    mappingRule.map = 'Observation.notReal';
+    unexpectedMapping.rules.push(mappingRule);
+
+    expect(mappings).toHaveLength(4); // Only the three mappings in the SD mapping list are created
```

Comment:
Four mappings are created, but the comment says "three mappings".

Reply:
Well oops. I should probably read my own comments when I update things.